file-formats/cfdf/type/zif_aff_cfdf_v1.intf.abap

```diff
@@ -0,0 +1,592 @@
+      "! $required
+      "! $maxLength 40
+      field_label   TYPE c LENGTH 40,
+
+      "! <p class="shorttext">Field Tooltip</p>
+      "! Field tooltip
+      "! $required
+      "! $maxLength 60
+      field_tooltip TYPE c LENGTH 60,
+    END OF ty_gs_ui_text.
+
+  TYPES: ty_gt_ui_text TYPE STANDARD TABLE OF ty_gs_ui_text WITH KEY language.
+
+  TYPES:
+    "! <p class="shorttext">Business Context Details</p>
+    "! Business context details
+    BEGIN OF ty_gs_business_context,
+      "! <p class="shorttext">Business Context</p>
+      "! Business context
+      "! $required
+      "! $maxLength 30
+      business_context     TYPE ty_business_context,
+
+      "! <p class="shorttext">Deviating Field Name</p>
+      "! Deviating field name
+      "! $maxLength 30
+      deviating_field_name TYPE ty_field_name,
+    END OF ty_gs_business_context.
+
+  TYPES: ty_gt_business_context TYPE STANDARD TABLE OF ty_gs_business_context WITH KEY business_context.
+
+  TYPES:
+    "! <p class="shorttext">Extended CDS View</p>
+    "! Extended CDS view
+    BEGIN OF ty_gs_extended_cds_view,
+      "! <p class="shorttext">CDS View Name</p>
+      "! CDS view name
+      "! $required
+      "! $maxLength 30
+      cds_view_name                 TYPE zif_aff_types_v1=>ty_object_name_30,
+
+      "! <p class="shorttext">Business Context</p>
+      "! Business context
+      "! $required
+      "! $maxLength 30
+      business_context              TYPE ty_business_context,
+
+      "! <p class="shorttext">Basic Search Relevance</p>
+      "! Basic search relevance
+      is_search_relevant            TYPE abap_bool,
+
+      "! <p class="shorttext">BAdI Implementation for Visibility Control</p>
+      "! BAdI implementation for visibility control
+      "! $maxLength 30
+      badi_impl_for_visibility_ctrl TYPE zif_aff_types_v1=>ty_object_name_30,
+
+      "! <p class="shorttext">Field Control Property</p>
+      "! Field control property
+      "! $maxLength 1
+      "! $values {@link zif_aff_cfdf_v1.data:co_field_control_property}
+      "! $default {@link zif_aff_cfdf_v1.data:co_field_control_property.optional}
+      field_control_property        TYPE ty_field_control_property,
+    END OF ty_gs_extended_cds_view.
+  TYPES: ty_gt_extended_cds_view TYPE STANDARD TABLE OF ty_gs_extended_cds_view WITH KEY business_context cds_view_name.
+
+  TYPES:
+    "! <p class="shorttext">Extended OData Service</p>
+    "! Extended OData service
+    BEGIN OF ty_gs_extended_odata_service,
+      "! <p class="shorttext">OData Service Key</p>
+      "! OData service key
+      "! $required
+      "! $maxLength 81
+      odata_service_key  TYPE c LENGTH 81,
+
+      "! <p class="shorttext">Business Context</p>
+      "! Business context
+      "! $required
+      "! $maxLength 30
+      business_context   TYPE ty_business_context,
+
+      "! <p class="shorttext">Basic Search Relevance</p>
+      "! Basic search relevance
+      is_search_relevant TYPE abap_bool,
```

Comment:
Question: Does "Basic" indicate that there are plans to provide different search mechanisms in the future, like "no relevance", "basic search", "any other search"?

If so, you might want to consider an enumeration here.

Reply:
No, this term should communicate the fact that the custom field should be considered in basic search of the corresponding Fiori UI, which is build on top of this OData service. So there won't be any changes here in future.